src/collection.js

```diff
@@ -186,3 +191,85 @@ function compareValues(a, b, descending) {
     return descending ? -1 : 1;
   }
 }
+
+function makeMapFn(projection) {
+  const id = projection && projection._id !== 0;
+  const keys = Object.getOwnPropertyNames(projection).filter(v => v !== '_id');
+  const passthroughfn = generatePassThroughFN(id, keys);
+
+  if (passthroughfn) {
+    return passthroughfn;
+  }
+
+  const type = incVsExc(projection);
+
+  if (type === 'inclusive') {
+    return function(d) {
+      const ret = {
+        _id: id ? d._id : undefined
+      };
+
+      for (const key of keys) {
+        ret[key] = d[key];
+      }
+
+      return ret;
+    };
+  } else if (type === 'exclusive') {
+    return function(d) {
+      const ret = {
+        _id: id ? d._id : undefined
+      };
+      for (const key in Object.getOwnPropertyNames(d)) {
+        if (!keys.includes(key)) {
+          ret[key] = d[key];
+        }
+      }
+
+      return ret;
+    };
+  } else {
+    return (d) => {
+      return d;
+    };
+  }
+}
+
+function generatePassThroughFN(id, keys) {
+  if (!keys.length && id) {
+    return function(d) {
+      return d;
+    };
+  } else if (!keys.length && !id) {
+    return function(d) {
+      const ret = {};
+      for (const key of keys) {
+        ret[key] = d[key];
+      }
+      return ret;
+    };
+  }
+}
+
+function incVsExc(projection) {
+  const values = [];
+
+  Object.entries(projection).forEach(([key, value]) => {
+    if (key !== '_id') {
+      values.push(value);
+    }
+  });
+
+  const ones = values.filter(v => v === 1);
```

Comment:
true and false are also supported, in addition to 0/1. So would be good to have `v === 1 || v === true` here